docs: add istio version compat table (#3268)
Kong/docs.konghq.com: app/kubernetes-ingress-controller/2.0.x/references/version-compatibility.md

| Kubernetes 1.20           | <i class="fa fa-check"></i> | <i class="fa fa-check"></i> | <i class="fa fa-check"></i> | <i class="fa fa-check"></i> | <i class="fa fa-check"></i> |
| Kubernetes 1.21           | <i class="fa fa-check"></i> | <i class="fa fa-check"></i> | <i class="fa fa-check"></i> | <i class="fa fa-check"></i> | <i class="fa fa-check"></i> |
| Kubernetes 1.22           | <i class="fa fa-times"></i> | <i class="fa fa-times"></i> | <i class="fa fa-times"></i> | <i class="fa fa-times"></i> | <i class="fa fa-check"></i> |

## Istio

> The {{site.kic_product_name}} can be integrated with an [Istio Service Mesh][istio] in order to use Kong Gateway as an ingress gateway for application traffic into the mesh network, an example of this is provided in our [Istio Guide][istio-guide].

Reviewer: ```suggestion
The {{site.kic_product_name}} can be integrated with an [Istio Service Mesh][istio] to use Kong Gateway as an ingress gateway for application traffic into the mesh network. See an example of this in the [Istio Guide][istio-guide].
```

> For each {{site.kic_product_name}} release tests are run to verify this documentation with upcoming versions of KIC and Istio: The result is the following table which highlights the combinations which have been tested:

Reviewer: ```suggestion
For each {{site.kic_product_name}} release, tests are run to verify this documentation with upcoming versions of KIC and Istio. The following table lists the tested combinations:
```
